lib/segment/client/http.ex

```diff
@@ -163,10 +163,12 @@ defmodule Segment.Http do
   defp prepare_events(items) when is_list(items), do: Enum.map(items, &prepare_events/1)
 
   defp prepare_events(item) do
-    Map.from_struct(item)
+    item
+    |> Map.from_struct()
     |> prep_context()
     |> add_sent_at()
     |> drop_nils()
+    |> add_timestamp()
```

Comment:
@stueccles I can close this PR and move the rest of the logic into my own app if you don't think it's the best direction for the library for you, but I'm curious what you think about this change. The idea here is that the timestamp is set by default to the s.entAt time if it's not set explicitly. This is the fallback I want for my analytics, but not sure if it's the best default for others. If you're into it I can cherry pick it into a new PR